Implementing the viewModel with hilt in order to link API and discoveryScreen

CassioManuguerra opened the pull request:
# PR
## README
This PR is done because of a bug from git that occurred after I renamed a branch on git and pushed twice locally. If you are curious for more information on what happened exactly contact me :)
## Changes
- The viewmodel is now supporting hilt
- Changed the OutdoorActivityRepository to an interface and a class extending it
- Started writing in the viewmodel for the functions to be linked
### Files changed : 
- DiscoveryScreen.kt
- AppModule.kt
- OutdoorActivityRepository.kt
- OutdoorActivityRepositoryIn.kt
- OutdoorActivityViewModel.kt

Thimphou reviewed (changes requested):
Few questions concerning your work:

- What does the DiscoveryScreenTest do? (It seems nothing, that's why I don't see the point of merging it to the main branch?)
- Hilt now supporting the viewmodel looks good to me.
- OutdoorActivityRepository is better as an interface, modularity, and code reusability will definitely be improved.
- Lastly, functions to be linked in the viewmodel do not seem to be finished? Maybe make some changes for these functions.

Otherwise, great job overall!

ecornamu commented:
Good structural changes ! Complying to the architecture of the projet is not an easy task and you seem to do just good ! Also changing `OutdoorActivityRepository` to an interface is smart and will help us with future reusability :) lgtm

CassioManuguerra commented:
# Update 
## Changes
- removed the unused test,
- added the main core for the API calling in Discovery Screen
- finished remodelling with Hilt
- Added a few comments

## What's next
- Fixing the map to work with the new OutdoorActivityRepository which now takes an APIservice as argument
- Fixing an API bug
